lighthouse-core/lib/sentry.js

```diff
@@ -3,94 +3,106 @@
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
  */
-// @ts-nocheck
 'use strict';
 
 const log = require('lighthouse-logger');
 
-// eslint-disable-next-line max-len
+/** @typedef {import('raven').CaptureOptions} CaptureOptions */
+/** @typedef {import('raven').ConstructorOptions} ConstructorOptions */
+
 const SENTRY_URL = 'https://a6bb0da87ee048cc9ae2a345fc09ab2e:63a7029f46f74265981b7e005e0f69f8@sentry.io/174697';
 
-const noop = () => Promise.resolve();
-const sentryApi = {
-  captureMessage: noop,
-  captureException: noop,
-  captureBreadcrumb: noop,
-  mergeContext: noop,
-  getContext: noop,
-};
+// Per-run chance of capturing errors (if enabled).
+const SAMPLE_RATE = 0.1;
 
 const SAMPLED_ERRORS = [
-  // Error code based sampling
-  {pattern: /DOCUMENT_REQUEST$/, rate: 0.1},
-  {pattern: /(IDLE_PERIOD|FMP_TOO_LATE)/, rate: 0.1},
-  {pattern: /^NO_.*/, rate: 0.1},
   // Message based sampling
-  {pattern: /Could not load stylesheet/, rate: 0.01},
-  {pattern: /Failed to decode/, rate: 0.1},
-  {pattern: /All image optimizations failed/, rate: 0.1},
-  {pattern: /No.*resource with given/, rate: 0.01},
-  {pattern: /No.*node with given id/, rate: 0.01},
+  {pattern: /Could not load stylesheet/, rate: 0.1},
+  {pattern: /No.*resource with given/, rate: 0.1},
+  {pattern: /No.*node with given id/, rate: 0.1},
```

Comment:
we could drop these completely? Also do they fire multiple times per run? We could fire once per audit instead (does sentry support adding a count payload and have a decent aggregation view of them?)